FreeRTOS/FreeRTOS-Kernel
# Add default implementations of vApplicationGetIdleTaskMemory and vApplicationGetTimerTaskMemory

Description
-----------
This PR introduces `configKERNEL_PROVIDED_STATIC_MEMORY` option which the application can set to 1 to use the default implementations of `vApplicationGetIdleTaskMemory` and `vApplicationGetTimerTaskMemory` functions.

If the application enables static allocation (i.e. sets `configUSE_STATIC_ALLOCATION` to 1) and does not provide the above 2
functions, it will result in linker error. The application has two options:

1. Set `configKERNEL_PROVIDED_STATIC_MEMORY` to 1 to use the default implementations of these functions.
2. Provide implementations of these 2 functions.

Note that default definitions are only available for non-MPU ports. The reason is that the stack alignment requirements vary for different architectures. 

Test Steps
-----------
Build a FreeRTOS project and set `configUSE_STATIC_ALLOCATION` to 1. Observe linker error for missing symbols  `vApplicationGetIdleTaskMemory` and `vApplicationGetTimerTaskMemory`. 

1. Set `configKERNEL_PROVIDED_STATIC_MEMORY` to 1. The linker error goes away and the default implementations in the tasks.c are used.
2. Set `configKERNEL_PROVIDED_STATIC_MEMORY` to 0 and provide implementation of `vApplicationGetIdleTaskMemory` and `vApplicationGetTimerTaskMemory`.  The linker error goes away and the default implementations are used.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
